core/src/androidMain/kotlin/com/ustadmobile/core/catalog/contenttype/VideoTypePluginAndroid.kt

```diff
@@ -7,162 +7,275 @@ import android.media.MediaFormat
+                            containerUid = repo.containerDao.insertAsync(this)
+                            contentJobItem.cjiContainerUid = containerUid
+                        }
+
+                db.contentJobItemDao.updateContainer(contentJobItem.cjiUid, container.containerUid)
+
+                val containerFolder = jobItem.contentJob?.toUri
+                        ?: defaultContainerDir.toURI().toString()
+                val containerFolderUri = DoorUri.parse(containerFolder)
+
+                if (compressVideo) {
+                    repo.addFileToContainer(container.containerUid, newVideo.toDoorUri(), newVideo.name,
+                            context,
+                            di,
+                            ContainerAddOptions(containerFolderUri))
+                } else {
+                    repo.addContainerFromUri(container.containerUid, localUri, context, di,
+                            localUri.getFileName(context),
+                            ContainerAddOptions(containerFolderUri))
+
+                    contentJobItem.cjiItemProgress = contentJobItem.cjiItemTotal / progressSize
+                    jobProgress.onProgress(contentJobItem)
+                }
+                videoTempDir.delete()
 
-                }, MediaTransformer.GRANULARITY_DEFAULT, null)
+                repo.addTorrentFileFromContainer(
+                        container.containerUid,
+                        DoorUri.parse(torrentDir.toURI().toString()),
+                        trackerUrl, containerFolderUri
+                )
 
+                val containerUidFolder = File(containerFolderUri.toFile(), container.containerUid.toString())
+                containerUidFolder.mkdirs()
+                ustadTorrentManager.addTorrent(container.containerUid, containerUidFolder.path)
 
-                try {
-                    videoCompleted.await()
-                } catch (e: Exception) {
-                    Napier.e(tag = VIDEO_ANDROID, throwable = e, message = e.message ?: "")
-                } finally {
-                    mediaTransformer.release()
+                contentJobItem.cjiConnectivityNeeded = true
+                db.contentJobItemDao.updateConnectivityNeeded(contentJobItem.cjiUid, true)
+
+                val haveConnectivityToContinueJob = checkConnectivityToDoJob(db, jobItem)
+                if (!haveConnectivityToContinueJob) {
+                    return@withContext ProcessResult(JobStatus.QUEUED)
+                }
+
+                val torrentFileBytes = File(torrentDir, "${container.containerUid}.torrent").readBytes()
+                uploadContentIfNeeded(contentNeedUpload, contentJobItem, jobProgress, httpClient, torrentFileBytes, endpoint)
+
+                repo.containerDao.findByUid(container.containerUid)
+
+                return@withContext ProcessResult(JobStatus.COMPLETE)
+            }catch(c: CancellationException){
+
+                withContext(NonCancellable){
```

Comment:
This is not canceling the media transformer... This could cause a problem two ways: 1) trying to delete a file whilst the media transformer is still working on it 2) allowing the media transformer to continue doing CPU intensive work for a long time after the user has said they want to cancel.